### PR TITLE
Feat: Multisig confirm update

### PR DESCRIPTION
**Pull request by egor0798**
## Changes:
- Split details for multisig operation card and mutisig confirmation modal window into 2 separate component
- Updated mutisig confirmation modal window
  - Added sender or account display
  - Added multisig deposit display and cross chain fee

***

image

### CODE REVIEW

File: src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx [hunk 1]
```diff
@@ -34,20 +52,46 @@ export const Confirmation = ({ tx, account, connection, feeTx }: Props) => {
...
+            className="text-footnote"
             api={connection.api}
             asset={connection.assets[0]}
             transaction={feeTx}
+            onFeeChange={(fee) => setIsFeeLoaded(Boolean(fee))}
           />
         )}
       </DetailRow>
+
+      {/*{isXcmTransaction(transaction) && xcmConfig && (*/}
```
Comment (egor0798) [inline src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx:78]: Will be updated later

File: src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx [hunk 2]
```diff
@@ -1,22 +1,51 @@
...
+
+  const asset = getAssetById(tx.transaction?.args.assetId, connection.assets) || connection.assets[0];
 
   const iconName = getIconName(tx.transaction);
+  const transaction =
+    tx.transaction?.type === 'batchAll'
+      ? tx.transaction.args.transactions.find(
+          (tx: Transaction) => tx.type === TransactionType.BOND || tx.type === TransactionType.UNSTAKE,
+        ) || tx.transaction.args.transactions[0]
+      : tx.transaction;
```
Comment (Asmadek) [inline src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx:44]: maybe make some improvements for this code? Move it to function (this code uses in few places), create variable for find value, use array.includes for type detection

File: src/renderer/pages/Operations/components/Details.tsx [hunk 3]
```diff
@@ -47,97 +53,68 @@ const Details = ({ tx, account, connection, isCardDetails = true }: Props) => {
...
-              <ChainTitle chainId={transaction?.args.destinationChain} fontClass={valueClass} />
-            </DetailRow>
-          )}
-        </>
-      )}
-
-      {transaction?.args.dest && (
-        <DetailRow label={t('operation.details.recipient')} className={valueClass}>
+      {account && (
+        <DetailRow label={t('operation.details.' + hasSender ? 'sender' : 'account')} className="text-text-secondary">
```
Comment (Asmadek) [inline src/renderer/pages/Operations/components/Details.tsx:82]:
```suggestion
        <DetailRow label={t(hasSender ? 'operation.details.sender' : 'operation.details.account')} className="text-text-secondary">
```
I think this way is more secure than string concatenation

**Review by Asmadek — APPROVED** [review]

File: src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx [hunk 4]
```diff
@@ -1,22 +1,51 @@
...
   tx: MultisigTransaction;
   account: MultisigAccount;
+  signatory?: Account;
   connection: ExtendedChain;
   feeTx?: Transaction;
+  onSign: () => void;
 };
-export const Confirmation = ({ tx, account, connection, feeTx }: Props) => {
+export const Confirmation = ({ tx, account, connection, signatory, feeTx, onSign }: Props) => {
+  const xcmConfig = useStore(sendAssetModel.$finalConfig);
```
Comment (tuul-wq) [inline src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx:32]: Better to use `useUnit` and move it 2 lines below
`const { t } = useI18n();` is a number 1 line in each component including translation 😉
```suggestion
  const xcmConfig = useUnit(sendAssetModel.$finalConfig);
```

File: src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx [hunk 5]
```diff
@@ -34,20 +63,45 @@ export const Confirmation = ({ tx, account, connection, feeTx }: Props) => {
...
             asset={connection.assets[0]}
             transaction={feeTx}
+            onFeeChange={(fee) => setIsFeeLoaded(Boolean(fee))}
           />
         )}
       </DetailRow>
+
+      {isXcmTransaction(transaction) && xcmConfig && (
+        <DetailRow label={t('operation.xcmFee')} className="text-text-primary pr-2">
+          {xcmConfig && connection.api && (
```
Comment (tuul-wq) [inline src/renderer/pages/Operations/components/ActionSteps/Confirmation.tsx:91]: `xcmConfig` has already been checked above - line `89`

File: src/renderer/pages/Operations/components/Details.tsx [hunk 6]
```diff
@@ -2,34 +2,40 @@ import cn from 'classnames';
...
   account?: MultisigAccount;
+  signatory?: Account;
   connection?: ExtendedChain;
-  isCardDetails?: boolean;
 };
 
-const Details = ({ tx, account, connection, isCardDetails = true }: Props) => {
+const Details = ({ tx, account, connection, signatory }: Props) => {
   const { t } = useI18n();
   const wallet = useUnit(walletModel.$activeWallet);
```
Comment (tuul-wq) [inline src/renderer/pages/Operations/components/Details.tsx:31]: `activeWallet` attracts more attention

File: src/renderer/pages/Operations/components/OperationCardDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+
+type Props = {
+  tx: MultisigTransaction;
+  account?: MultisigAccount;
+  connection?: ExtendedChain;
+};
+
+const OperationCardDetails = ({ tx, account, connection }: Props) => {
+  const { t } = useI18n();
+  const wallet = useUnit(walletModel.$activeWallet);
```
Comment (tuul-wq) [inline src/renderer/pages/Operations/components/OperationCardDetails.tsx:26]: Same here
Checked other files, we stick to `activeWallet`

File: src/renderer/pages/Operations/components/OperationCardDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+                `${blockCreated}-${indexCreated}`
+              )}
+            </DetailRow>
+          )}
+        </>
+      )}
+    </dl>
+  );
+};
+export default OperationCardDetails;
```
Comment (tuul-wq) [inline src/renderer/pages/Operations/components/OperationCardDetails.tsx:262]: Prefer named export

File: src/renderer/pages/Operations/components/OperationFullInfo.tsx [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import { Icon, Button, InfoLink, SmallTitleText } from '@renderer/shared/ui';
-import Details from '@renderer/pages/Operations/components/Details';
+import OperationCardDetails from '@renderer/pages/Operations/components/OperationCardDetails';
```
Comment (tuul-wq) [inline src/renderer/pages/Operations/components/OperationFullInfo.tsx:2]: Import can be shortened

File: src/renderer/pages/Operations/components/modals/ApproveTx.tsx [hunk 10]
```diff
@@ -231,16 +231,14 @@ const ApproveTx = ({ tx, account, connection }: Props) => {
         onClose={handleClose}
       >
         {activeStep === Step.CONFIRMATION && (
-          <>
```
Comment (tuul-wq) [inline src/renderer/pages/Operations/components/modals/ApproveTx.tsx:234]: 👍

File: src/renderer/shared/ui/DetailRow/DetailRow.tsx [hunk 11]
```diff
@@ -19,7 +19,7 @@ const DetailRow = ({ label, className, children }: PropsWithChildren<Props>) =>
     )}
 
     {typeof children === 'string' ? (
-      <FootnoteText as="dd" align="right" className={cn(className, 'justify-end py-[3px] px-2')}>
+      <FootnoteText as="dd" align="right" className={cn('justify-end py-[3px] px-2', className)}>
```
Comment (tuul-wq) [inline src/renderer/shared/ui/DetailRow/DetailRow.tsx:22]: 👍 
can also change to `cnTw`

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/entities/multisig/lib/multisigTx/common/utils.ts [hunk 12]
```diff
@@ -189,3 +189,15 @@ export const buildMultisigTx = (
     transaction,
   };
 };
+
+export const getTransactionFromMultisigTx = (tx: MultisigTransaction): Transaction | DecodedTransaction | undefined => {
```
Comment (tuul-wq) [inline src/renderer/entities/multisig/lib/multisigTx/common/utils.ts:193]:
```javascript
export const getTransactionFromMultisigTx = (tx: MultisigTransaction): Transaction | DecodedTransaction | undefined => {
  if (!tx.transaction || tx.transaction.type !== 'batchAll') {
    return tx.transaction;
  }

  const transactionMatch = tx.transaction.args.transactions.find((tx: Transaction) => {
    return tx.type === TransactionType.BOND || tx.type === TransactionType.UNSTAKE;
  });

  return transactionMatch || tx.transaction.args.transactions[0];
};
```

**Review by tuul-wq — CHANGES_REQUESTED** [review]
One code style change

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**